libevmasm/AssemblyItem.cpp

```diff
@@ -40,13 +42,15 @@ AssemblyItem AssemblyItem::toSubAssemblyTag(size_t _subId) const
 pair<size_t, size_t> AssemblyItem::splitForeignPushTag() const
 {
 	assertThrow(m_type == PushTag || m_type == Tag, Exception, "");
-	return make_pair(size_t((data()) / (u256(1) << 64)) - 1, size_t(data()));
+	// There may be a bug in boost where typecasting u256 to size_t results in all bits set if the higher bit was set
+	return make_pair(size_t((data() >> 64) - 1), size_t(data() & 0xffffffffffffffff));
 }
 
 void AssemblyItem::setPushTagSubIdAndTag(size_t _subId, size_t _tag)
 {
 	assertThrow(m_type == PushTag || m_type == Tag, Exception, "");
-	setData(_tag + (u256(_subId + 1) << 64));
+	assertThrow(_tag < (u256(1) << 64), Exception, "");
+	setData(_tag + ((u256(_subId) + 1) << 64));
```

Comment:
I think `_subId + 1` was used on purpose, because `subId == -1` in some cases (for example in the case of "current sub id").

Reply:
So it might be that this could have been the root cause: depending on the system, `size_t(-1) + 1` is not always equal to `0`...

Reply:
(because the type of the literal 1 is larger than the type of `size_t`.